src/prop-validators/arrayWithLength.js

```diff
@@ -0,0 +1,59 @@
+import propTypes from 'prop-types'
+
+const arrayWithLengthFactory = ({
+    min = 0,
+    max = Infinity,
+    propType,
+    isRequired,
+}) => (props, propName, componentName) => {
+    const arr = props[propName]
+
+    if (isRequired && typeof arr === 'undefined') {
+        return new Error(`${propName} is required.`)
+    }
+
+    if (arr && !Array.isArray(arr)) {
+        return new Error(`${propName} is not an array.`)
+    }
+
+    if (arr && arr.length > max) {
+        return new Error(
+            // prettier-ignore
+            `${propName} array has a length of ${arr.length}, but the maximum is ${max}`
+        )
+    }
+
+    if (arr && arr.length < min) {
+        return new Error(
+            // prettier-ignore
```

Comment:
Why are the ignores necessary?

Reply:
I added these ignore comments because I thought that the way that prettier is breaking up these template strings makes them very hard to read. Prettier doesn't have a setting to change its behaviour in this regard https://github.com/prettier/prettier/issues/3280#issuecomment-345105820

We can remove it if you prefer, or I could try to write a more succinct message.

Reply:
For illustration, this is how prettier's formatting looks:
```javascript
`${propName} array has a length of ${
    arr.length
}, but the maximum is ${max}`
```

Reply:
Not too bad, I was wondering if there was a serious problem. You can keep the ignores.